tensorflow_quantum/core/src/util_qsim.h

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2020 The TensorFlow Quantum Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef UTIL_QSIM_H_
+#define UTIL_QSIM_H_
+
+#include <cstdint>
+#include <vector>
+
+#include "tensorflow/core/framework/op_kernel.h"
+
+namespace tfq {
+
+// Custom FOR loop struct to use TF threadpool instead of native
+// qsim OpenMP or serial FOR implementations.
+struct QsimFor {
+  tensorflow::OpKernelContext* context;
+  QsimFor(tensorflow::OpKernelContext* cxt) { context = cxt; }
+
+  template <typename Function, typename... Args>
```

Comment:
Looks nice! I didn't know how to make varied length arguments in template. Thank you!